lib/puppet/provider/swap_file/linux.rb

```diff
@@ -0,0 +1,110 @@
+Puppet::Type.type(:swap_file).provide(:linux) do
+
+  confine  :kernel   => :linux
+  commands :swapon   => 'swapon'
+  commands :swapoff  => 'swapoff'
+  commands :mkswap   => 'mkswap'
+
+  mk_resource_methods
+
+  def initialize(value={})
+    super(value)
+    @property_flush = {}
+  end
+
+  def self.get_swap_files
+    swapfiles = swapon(['-s']).split("\n")
+    swapfiles.shift
+    swapfiles.sort
+  end
+
+  def self.prefetch(resources)
+    instances.each do |prov|
+      if resource = resources[prov.name]
+        resource.provider = prov
+      end
+    end
+  end
+
+  def self.instances
+    get_swap_files.collect do |swapfile_line|
+      swapfile_properties = get_swapfile_properties(swapfile_line)
```

Comment:
You don't really need the intermediary var here `new(get_swapfile_properties(swapfile_line))`